### ISSUE TITLE
[a11y]: TagOverflow incorrect role

### ISSUE THREAD

**Issue by naorunaoru** [carbon-design-system/ibm-products#5964]
### Package

@carbon/ibm-products

### Package version

2.47.0

### Automated testing tool and ruleset

IBM Equal Access Checker

### Description

The `main` landmark role is used to indicate the primary content of a document. It should be removed from the component. 

### Reproduction/example

https://carbon-for-ibm-products.netlify.app/?path=/docs/experimental-components-tag-overflow-tagoverflow--docs

### Steps to reproduce

Follow the source code or run EAC extension

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by amal-k-joy** [comment]
Looks like `role="main"` is added in the component level
